### ISSUE TITLE
o2p Hyperparameter Sweep

### ISSUE THREAD

**Issue by sharkeyboi** [emelex-ai/BRIDGE#183]
- [ ] Implement metrics logging for test set data

- [ ] Do hyperparameter sweep on learning rate, batch size using test metrics.
